x-pack/plugins/cases/public/common/lib/kibana/hooks.ts

```diff
@@ -160,23 +160,48 @@ export const useNavigation = (appId: string) => {
   return { navigateTo, getAppUrl };
 };
 
+interface Capabilities {
+  crud: boolean;
+  read: boolean;
+}
+interface UseApplicationCapabilities {
+  actions: Capabilities;
+  generalCases: Capabilities;
+  visualize: Capabilities;
+  dashboard: Capabilities;
+}
+
 /**
- * Returns the capabilities of the main cases application
+ * Returns the capabilities of various applications
  *
  */
-export const useApplicationCapabilities = (): { crud: boolean; read: boolean } => {
-  const capabilities = useKibana().services.application.capabilities;
-  const casesCapabilities = capabilities[FEATURE_ID];
-  return {
-    crud: !!casesCapabilities?.crud_cases,
-    read: !!casesCapabilities?.read_cases,
-  };
-};
-export const useKibanaCapabilities = (): { visualize?: boolean; dashboard?: boolean } => {
+
+export const useApplicationCapabilities = (): UseApplicationCapabilities => {
   const capabilities = useKibana().services?.application?.capabilities;
+  const casesCapabilities = capabilities[FEATURE_ID];
 
-  return {
-    visualize: !!capabilities?.visualize?.save,
-    dashboard: !!capabilities?.dashboard?.show && !!capabilities?.dashboard?.createNew,
-  };
+  return useMemo(
+    () => ({
+      actions: { crud: !!capabilities.actions?.save, read: !!capabilities.actions?.show },
+      generalCases: {
+        crud: !!casesCapabilities?.crud_cases,
+        read: !!casesCapabilities?.read_cases,
+      },
+      visualize: { crud: !!capabilities.visualize?.save, read: !!capabilities.visualize?.show },
```

Comment:
I'm not a fan of proxying values like this. Because we can simply just put this code in the place we need it and that's it. That's why I kept visualize as a single boolean value that checks both conditions instead of separate ones.

Reply:
There are scenarios in which you need to know if the user has `read` or `crud` access and not both. Also, as a consumer of the hook if I have a single boolean I am not sure what kind of permissions the user has. I have to look at the code of the hook. By having separate properties I can immediately understand what is going on, it leaves no ambiguity and avoids security mistakes.

Reply:
Each plugin has a different name to define `crud` or `read` operations. As a developer, I need to know every time what this property is called and what is about. Also, the capabilities object is untyped.  Having the hook add typing and there is no need to know the names. It will always be `crud` and `read`. Also, some of them have more than two capabilities. For example, the actions plugin has `read`, `execute`, `delete`, `save`. Surprisingly `read`, `execute` are both considered as `read`. A new developer may make a mistake and use the wrong one. For all that reasons, I think is fine having this hook that a) force types b) export the same contract c) centralize all cases dependencies regarding capabilities.

Reply:
I was not aware the capabilities was totally untyped. that's a good point.  I see your point then as using this for standardizing them.